lib/intl.ts

```diff
@@ -214,14 +324,16 @@ function amend(optionsParam = {}, amended = {}) {
     'timeZoneName',
     'dateStyle',
     'timeStyle'
-  ]) {
-    options[opt] = opt in amended ? amended[opt] : options[opt];
-    if (options[opt] === false || options[opt] === undefined) delete options[opt];
+  ] as const) {
+    // TODO: can this be typed more cleanly?
+    type OptionMaybeFalse = typeof options[typeof opt] | false;
```

Comment:
Hopefully there aren't any boolean-typed options added to Intl.DateTimeFormat that default to true... :|